src/Assets/AssetContainer.php

```diff
@@ -170,16 +174,55 @@ public function blueprint()
         return $blueprint;
     }
 
+    public function afterSave($callback)
+    {
+        $this->afterSaveCallbacks[] = $callback;
+
+        return $this;
+    }
+
+    public function saveQuietly()
+    {
+        $this->withEvents = false;
+
+        $result = $this->save();
+
+        $this->withEvents = true;
+
+        return $result;
+    }
+
     /**
      * Save the container.
      *
      * @return void
      */
     public function save()
     {
+        $isNew = is_null(Facades\AssetContainer::find($this->id()));
```

Comment:
@jasonvarga I've copied this `$isNew` thing from the `Entry` class into all of these other classes. 

However, for whatever reason, this always seems to be return `true`, which causes my `SomethingCreated` events to fire more than once when saving.

Do you have any ideas on how to fix that? It's the only real blocker I have between this being a draft and a PR ready for review and I'm not quite sure what to do.

*I would link the Action results but that seems to have disappeared*

Reply:
@duncanmcclean take a closer look at how `EntryTest->it_dispatches_entry_created_only_once()` is written. In line 657 it says how the `Entry` facade should respond to the consecutive `find()` calls:

```php
        Facades\Entry::shouldReceive('find')->with('1')->times(3)->andReturn(null, $entry, $entry);
```

You need something similar in your tests, for example your `BlueprintTest->it_dispatches_blueprint_created_only_once()` succeeds with the following code:

```php
    public function it_dispatches_blueprint_created_only_once()
    {
        Event::fake();

        $blueprint = new Blueprint;

        Facades\Blueprint::shouldReceive('save')->with($blueprint);
        Facades\Blueprint::shouldReceive('find')->with($blueprint->handle())->times(3)->andReturn(null, $blueprint, $blueprint);

        $blueprint->save();
        $blueprint->save();
        $blueprint->save();

        Event::assertDispatched(BlueprintSaved::class, 3);
        Event::assertDispatched(BlueprintCreated::class, 1);
    }
```

Reply:
Ah, that makes sense. I don't think I would have thought of that myself 😅